src/lib/components/Filter/Filter.js

```diff
@@ -297,35 +301,48 @@ export class Filter extends Component {
     // Build new component state or retrieve it from FILTER store
     const filterState = nextProps.FILTER[layerId];
     const layerFilters = this.getLayerFilter(layerId); // this may be deprecated
-    const filterOptions = filterState && filterState.filterOptions
-      ? filterState.filterOptions
-      : (nextProps.timeseriesObj && layerObj.aggregate && layerObj.aggregate.timeseries)
-        ? generateFilterOptions(timeseriesObj) : (layerObj.filterOptions || {});
 
-    const filters = (filterState && filterState.filters)
-      || (this.state.isFiltered && this.state.prevFilters)
-      || this.buildFiltersMap(filterOptions, layerFilters, this.state.prevFilters);
+    let filterOptions;
+    if (nextProps.timeseriesObj && layerObj.aggregate && layerObj.aggregate.timeseries) {
+      filterOptions = generateFilterOptions(timeseriesObj);
+    } else if (!timeseriesObj && (filterState && filterState.filterOptions)) {
+      filterOptions = filterState.filterOptions;
+    } else {
+      filterOptions = (layerObj.filterOptions || {})
+    }
+
+    let filters;
 
+    if (timeseriesObj &&
+      (timeseriesObj.temporalIndex !== (this.props.timeseriesObj &&
+        this.props.timeseriesObj.temporalIndex)) && !layerFilters &&
+      !(filterState && filterState.doUpdate)) {
+      filters = this.buildFiltersMap(filterOptions, layerFilters, this.state.prevFilters)
+    } else {
+      filters = (filterState && filterState.filters) ||
+        (this.state.isFiltered && this.state.prevFilters) ||
+        this.buildFiltersMap(filterOptions, layerFilters, this.state.prevFilters)
+    }
 
     // determine whether to update the compnent state
-    const doUpdate = (layerId !== this.state.layerId
-      && (filterOptions && Object.keys(filterOptions).length > 0))
-      || (filterState && filterState.doUpdate);
+    const doUpdate = ((layerId !== this.state.layerId &&
+        (filterOptions && Object.keys(filterOptions).length > 0)) ||
+      (filterState && filterState.doUpdate) || (timeseriesObj &&
+        (timeseriesObj.temporalIndex !== (this.props.timeseriesObj &&
+          this.props.timeseriesObj.temporalIndex))));
```

Comment:
@KipSigei it looks like this last condition will fail when on the first time period because the `temporalIndex` will be `0` which is falsey. Is this intentional or should it check if `typeof temporalIndex === 'undefined'`?

Reply:
It won't reach that point because temproralIndex is a string not integer, so `"0"` wont be falsey @cKellyDesign

Reply:
perfect!